rules/filename-case.js

```diff
@@ -78,6 +82,11 @@ const create = context => {
 		Program: node => {
 			const extension = path.extname(filenameWithExt);
 			const filename = path.basename(filenameWithExt, extension);
+
+			if (isIndexFile(filenameWithExt)) {
```

Comment:
I would just do `if (filename === 'index.js') {` and remove the `isIndexFile` function.